src/hashmap.c

```diff
@@ -493,3 +493,25 @@ ssize_t hashmap_remove (hashmap_t map, const char *key)
         /* The key was not found, so return 0 */
         return deleted;
 }
+
+/*
+ * Look up the value for a variable.
+ */
+char *lookup_variable (hashmap_t map, const char *varname)
```

Comment:
I am working on a personal project that involves hacking tinyproxy to log user-agent strings in addition to other minor tweaks. This means I need to do hash lookups from reqs.c to create the appropriate log messages. I moved lookup_value() from html-error.c to hashmap.c so that it could be more generally useful. In this way hashmap.c provides a more complete interface for any module seeking to create and use hashmaps.

If this is desired in addition to the OSX build fixes, I can split this into a separate PR with two patches.